src/py42/clients/cases.py

```diff
@@ -0,0 +1,196 @@
+            findings=findings,
+        )
+
+    @staticmethod
+    def _make_range(begin_time, end_time):
+        if not begin_time and not end_time:
+            return None
+        if not begin_time:
+            begin_time = datetime.utcfromtimestamp(0)
+        if not end_time:
+            end_time = datetime.utcnow()
+        end = parse_timestamp_to_milliseconds_precision(end_time)
+        start = parse_timestamp_to_milliseconds_precision(begin_time)
+        return "{}/{}".format(start, end)
+
+    def get_all(
+        self,
+        name=None,
+        status=None,
+        min_create_time=None,
+        max_create_time=None,
+        min_update_time=None,
+        max_update_time=None,
+        subject=None,
+        assignee=None,
+        page_number=1,
+        page_size=100,
+        sort_direction=u"asc",
+        sort_key=u"number",
+        **kwargs
+    ):
+        """Gets all cases.
+
+        Args:
+            name (str, optional): Filter results by case name, matches partial names. Defaults to None.
+            status (str, optional): Filter results by case status. `OPEN` or `CLOSED` Defaults to None.
+            min_create_time (str or int or float or datetime, optional): Filter results by case creation time, start time.
+                 str format %Y-%m-%d %H:%M:%S. Defaults to None.
+            max_create_time (str or int or float or datetime, optional): Filter results by case creation time, end time.
+                 str format %Y-%m-%d %H:%M:%S. Defaults to None.
+            min_update_time (str or int or float or datetime, optional): Filter results by last updated time, start time.
+                 str format %Y-%m-%d %H:%M:%S. Defaults to None.
+            max_update_time (str or int or float or datetime, optional): Filter results by last updated time, end time.
+                 str format %Y-%m-%d %H:%M:%S. Defaults to None.
+            subject (str, optional): Filter results based on User UID of a subject of a case. Defaults to None.
+            assignee (str, optional): Filter results based on User UID of an assignee of a case. Defaults to None.
+            page_number (int, optional): Page number of the results. Defaults to 1.
+            page_size (int, optional): Number of results to return per page. Defaults to 100.
+            sort_direction (str, optional): The direction on which to sort the response,
+                based on the corresponding sort key. `asc` or `desc`. Defaults to `asc`.
+            sort_key (str, optional): Values on which the response will be sorted. Defaults to "number".
+                Available options are `name`, `number`, `createdAt`, `updatedAt`, `status`, `assigneeUsername`, `subjectUsername`.
+
+        Returns:
+            generator: An object that iterates over :class:`py42.response.Py42Response` objects
+            that each contain a page of cases.
+        """
+        if status and status.upper() not in CaseStatus.choices():
```

Comment:
There is no need to do this... This should be handled on the server.  What if they add another status?  Now we have to go update py42... Let the user pass in a `str` for the value... We provide enums as a guidance, not as a the law.

Reply:
That is the reason why I was hesitant to use it, so now if we have the enum and user passes some another string and we allow that then what is the point of using enum?

Reply:
the point is convenience -- it makes it so that the user doesn't have to memorize the list available strings for this parameter since they are available in code.

This does mean that if a new value becomes possible that a user _would_ have to just know it in order to use it, but that is better than having them try it our method erroring.

Reply:
I beg to differ, we have documentation to help so that they don't have to memorize. 
However its 2 against 1, so I will keep the enum and allow string.

Reply:
Changes done.